common.py

```diff
@@ -8,6 +8,22 @@
 TFT_TESTS = "tft-tests"
 
 
+def enum_factory(enum_type: Type[Enum]):
+    def factory(value: Any) -> Enum:
```

Comment:
granted, the function handles `Any` type (to some definition of the work "handle"). But in practice, it only accepts strings or the enum type.

Wouldn't it be better, to tell the type checker about that, and return a `value: str | T` value (where `T` is the Enum type)? Because passing anything else is not going to work, and likely a bug we want to be told by the type checker.

Reply:
Good suggestion, I added that